etcdctl/command/cluster_health.go

```diff
@@ -70,51 +71,60 @@ func handleClusterHealth(c *cli.Context) {
 
 	for {
 		health := false
+		wait := sync.WaitGroup{}
+		wait.Add(len(ms))
 		for _, m := range ms {
-			if len(m.ClientURLs) == 0 {
-				fmt.Printf("member %s is unreachable: no available published client urls\n", m.ID)
-				continue
-			}
-
-			checked := false
-			for _, url := range m.ClientURLs {
-				resp, err := hc.Get(url + "/health")
-				if err != nil {
-					fmt.Printf("failed to check the health of member %s on %s: %v\n", m.ID, url, err)
-					continue
-				}
+			go func(m client.Member) {
```

Comment:
This is getting a bit too nested IMHO; can we split it out into 
`checkMemberHealth(hc http.Client, m client.Member)`

```
for _, m := range ms {
     go func(m client.Member) {
        checkMemberHealth(hc, m)
        wg.Done()
    }(m)
}
```
Then you can also eliminate the `checked` variable and return instead of breaking.